### ISSUE TITLE
[💡FEATURE REQUEST]: Current impl of the `worker_watcher` queue container may violate `Pop` <-> `Push` operation under some circumstances

### ISSUE THREAD

**Issue by rustatian** [roadrunner-server/roadrunner#750]
### Edit: Fixed in the 2.4.0 for the worker's container based on the channel. As the target of this ticket, we need to create a new container described above.

Under some circumstances (TTL, supervisor), when using custom plugins, the order of `Pop` and `Push` operations may be violating. When Pool takes the Worker, another worker, which waits on the channel mutex might be pushed into the channel. So, the `Pop` <-> `Push` relation might be broken and the worker won't be able to return a response.
Another issue here is if the user used TTL without requests, the killed processes will be on channel mutex:
```go
func (v *Vec) Push(w worker.BaseProcess) {
	v.workers <- w
}
```
We need to do the following:
1. Remove the duplicated slice with processes pointers from the `worker_watcher`.
2. Implement the `Queue` data structure which should support the following operations (all O(1)): `Pop`, `Push`, `Lish` (O(n)), `Remove`, `Destroy`. This data structure should also support the `allocate_timeouts`.
3. Integrate this data structure with `worker_watcher`.
4. `Remove` operation should be used to remove worker by its PID from the `Queue` without stacking dead processed on the mutex. The dead worker might be at any position in the `Queue`.
4. Rethink `List()` operation. We don't need to return a full process, but only the Memory, CPU, Execs, Time parts.

**Comment by rustatian** [comment]
fixed by: https://github.com/roadrunner-server/sdk/pull/36
